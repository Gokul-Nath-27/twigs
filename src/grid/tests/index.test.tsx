import React from 'react';
import 'jest';
import '@testing-library/jest-dom';
import { createRoot } from 'react-dom/client';
import renderer, { act } from 'react-test-renderer';
import { Grid } from '../index';

describe('Grid', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const root = createRoot(div!);
    act(() => {
      root.render(<Grid templateColumns="1fr 1fr 1fr" />);
      root.unmount();
    });
  });

  it('renders with width prop', () => {
    const div = document.createElement('div');
    const root = createRoot(div!);
    act(() => {
      root.render(<Grid width={350} templateColumns="1fr 1fr 1fr" />);
      root.unmount();
    });
  });

  it('matches snapshot: renders with templatearea props', () => {
    const tree = renderer.create(
      <Grid
        templateAreas={`"navbar navbar"
                        "sidebar main"`}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: renders with templatearea and templateColumns props', () => {
    const tree = renderer.create(<Grid
      templateAreas={`"navbar navbar"
                        "sidebar main"`}
      templateColumns="342px 1fr"
    />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: renders with templatearea and templateRows props', () => {
    const tree = renderer.create(<Grid
      templateAreas={`"navbar navbar"
        "sidebar main"`}
      templateRows="64px 1fr"
    />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: renders with templatearea, templateRows and templateColumns props', () => {
    const tree = renderer.create(<Grid
      templateAreas={`"navbar navbar"
          "sidebar main"`}
      templateRows="64px 1fr"
      templateColumns="342px 1fr"
    />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: renders with templatearea, templateColumns props and repeat props', () => {
    const tree = renderer.create(<Grid
      templateAreas={`"navbar navbar"
                        "sidebar main"`}
      templateColumns="342px 1fr"
      repeat="fit"
    />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: renders with templatearea, templateRows, templateColumns, gap and custom styles', () => {
    const tree = renderer.create(
      <Grid
        templateAreas={`"navbar navbar"
                         "sidebar main"`}
        templateRows="64px 1fr"
        templateColumns="342px 1fr"
        gap={[0, 0]}
        css={{
          height: '100vh',
          width: '100vw',
        }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
