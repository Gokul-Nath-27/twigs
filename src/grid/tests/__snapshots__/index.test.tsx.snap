// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid matches snapshot: renders with templatearea and templateColumns props 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-idPnhof-css"
  data-testid="grid"
/>
`;

exports[`Grid matches snapshot: renders with templatearea and templateRows props 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-ijqqdFk-css"
  data-testid="grid"
/>
`;

exports[`Grid matches snapshot: renders with templatearea props 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-icMpRbk-css"
  data-testid="grid"
/>
`;

exports[`Grid matches snapshot: renders with templatearea, templateColumns props and repeat props 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-idPnhof-css"
  data-testid="grid"
/>
`;

exports[`Grid matches snapshot: renders with templatearea, templateRows and templateColumns props 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-icoXYID-css"
  data-testid="grid"
/>
`;

exports[`Grid matches snapshot: renders with templatearea, templateRows, templateColumns, gap and custom styles 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-ijwHXIl-css"
  data-testid="grid"
/>
`;
